jupyterhub/app.py

```diff
@@ -1975,25 +1975,63 @@ async def init_groups(self):
                 group.users.append(user)
         db.commit()
 
-    async def init_roles(self):
+    async def init_role_creation(self):
         """Load default and predefined roles into the database"""
-        db = self.db
-        # tokens are added separately
-        role_bearers = ['users', 'services', 'groups']
-
-        # load default roles
         self.log.debug('Loading default roles to database')
         default_roles = roles.get_default_roles()
-        for role in default_roles:
-            roles.create_role(db, role)
+        config_role_names = [r['name'] for r in self.load_roles]
+        init_roles = self.load_roles.copy()
+        for role_name in config_role_names:
+            if config_role_names.count(role_name) > 1:
+                raise AttributeError(
```

Comment:
```suggestion
                raise ValueError(
```

ValueError is the more appropriate error for when the value of something is a problem. AttributeError is generally reserved for attempting to access an undefined attribute.

More information on the [Python exceptions list](https://docs.python.org/3/library/exceptions.html).